### PR TITLE
v8.0.0rc1

### PR DESCRIPTION
**Pull request by leofang**
Testing multiple local runners.

### CODE REVIEW

**Comment by leofang** [comment]
Confirmed two local runners running in parallel!

截圖 2020-08-28 上午2 11 35

**Comment by mrakitin** [comment]
I converted the PR to draft, please make it "ready for review" whenever ready. Thanks!

What about the sequence of channels? In another package we agreed to have `nsls2forge,defaults,conda-forge`, but here `conda-forge` goes second...

**Comment by leofang** [comment]
The CUDA 10.1 docker image needs to be fixed (https://github.com/conda-forge/docker-images/pull/154).

**Comment by leofang** [comment]
Power-cycle to use the fixed CUDA images.

**Comment by jakirkham** [comment]
Please let us know how it goes 🙂

**Comment by leofang** [comment]
@jakirkham It works! Thanks for quickly fixing it.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
+{% set version = "8.0.0rc1" %}
+{% set sha256 = "2981c809f7a7e1964c83921bcc065ae573e019f48f64468dc345bb3cd451e84e" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  - url: https://github.com/cupy/cupy/archive/v{{ version }}.tar.gz
+  - url: https://pypi.io/packages/source/c/cupy/cupy-{{ version }}.tar.gz
```
Comment (mrakitin) [inline recipe/meta.yaml]: We like parametrization of such cases :)
```suggestion
  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,31 +1,22 @@
...
   # temporarily disable PyPy build (conda-forge/cupy-feedstock#43)
   skip: true  # [python_impl == "pypy"]
   skip: true  # [cuda_compiler_version in (undefined, "None")]
 
   script:
-    # The issue of some CUDA headers being moved out of /usr/local/cuda/ since 10.1+
-    # is addressed in conda-forge/docker-images#134, so we do nothing in the recipe
-    - export CUB_PATH=${PWD}/cub_v1.8.0  # TODO(leofang): remove this
     #- export CUTENSOR_PATH=${HOME}/libcutensor  # [cuda_compiler_version in ('10.1', '10.2')]
     - {{ PYTHON }} -m pip install . -vv
```
Comment (mrakitin) [inline recipe/meta.yaml]: As recommended in https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip:
```suggestion
    - {{ PYTHON }} -m pip install . --no-deps -vv
```

**Review by mrakitin — COMMENTED** [review]
Looks good. Left a couple of style suggestions.

**Comment by mrakitin** [comment]
@leofang, is it unavoidable to have these 2 dependencies?
```
    nccl:                   2.7.8.1-h7cc98d6_0      conda-forge
    nvcc_linux-64:          9.2-h7cc98d6_6          conda-forge
```
They may not exist in `defaults`, and the resulted package may not be compatible with the rest of our stack.

File: recipe/meta.yaml (hunk 1 above)
Comment (leofang) [reply recipe/meta.yaml]: Thanks!

**Comment by leofang** [comment]
> is it unavoidable to have these 2 dependencies?
> 
> ```
>     nccl:                   2.7.8.1-h7cc98d6_0      conda-forge
>     nvcc_linux-64:          9.2-h7cc98d6_6          conda-forge
> ```
> 
> They may not exist in `defaults`, and the resulted package may not be compatible with the rest of our stack.

Hi @mrakitin, I think you caught them in the test environment? For `nvcc_linux-64` it's a good catch --- for some reason it shows up in the CI along with the compilers (`gcc_linux-64` and `gxx_linux-64`). I just checked and it shows up in the conda-forge builds as well. But when actually installing the package none of these would be pulled, so this only has to do with the complex `run_export` internal applied to the compilers. (@jakirkham, any chance you have insight on this?)

As for nccl, it's not available on default unfortunately. It's NVIDIA's GPU communication library providing certain MPI-like operations. I think we'll have to stick around with it. For example our ptychography code could use nccl. My impression is NVIDIA will gradually take over the control of CUDA-related packages, and in the future it may or may not be from conda-forge.

**Comment by leofang** [comment]
Let me merge this, fix https://github.com/conda-forge/cupy-feedstock/pull/71, and fix #5.

**Pull request merged**